lib/AST/ASTScopeCreation.cpp

```diff
@@ -0,0 +1,1061 @@
+    createSubtree<SpecializeAttributeScope>(specializeAttr, decl);
+  });
+
+  // Create scopes for generic and ordinary parameters.
+  // For a subscript declaration, the generic and ordinary parameters are in an
+  // ancestor scope, so don't make them here.
+  ASTScopeImpl *leaf = this;
+  if (!isa<AccessorDecl>(decl)) {
+    leaf = createGenericParamScopes(decl, decl->getGenericParams());
+    if (!decl->isImplicit()) {
+      leaf = leaf->createSubtree<AbstractFunctionParamsScope>(
+          decl->getParameters(), nullptr);
+    }
+  }
+  // Create scope for the body.
+  if (decl->getBody()) {
+    if (decl->getDeclContext()->isTypeContext())
+      leaf->createSubtree<MethodBodyScope>(decl);
+    else
+      leaf->createSubtree<PureFunctionBodyScope>(decl);
+  }
+}
+
+void AbstractFunctionBodyScope::expandMe(DeferredNodes &deferred) {
+  BraceStmt *braceStmt = decl->getBody();
+  ScopeCreator().visitBraceStmt(braceStmt, this, deferred);
+}
+
+void PatternEntryDeclScope::expandMe(DeferredNodes &deferred) {
+  auto patternEntry = getPatternEntry();
+  // Create a child for the initializer, if present.
+  // Cannot trust the source range given in the ASTScopeImpl for the end of the
+  // initializer (because of InterpolatedLiteralStrings and EditorPlaceHolders),
+  // so compute it ourselves.
+  SourceLoc initializerEnd;
+  if (patternEntry.getInitAsWritten() &&
+      patternEntry.getInitAsWritten()->getSourceRange().isValid()) {
+    auto *initializer =
+        createSubtree<PatternEntryInitializerScope>(decl, patternEntryIndex);
+    initializerEnd = initializer->getSourceRange().End;
+  }
+  // If there are no uses of the declararations, add the accessors immediately.
+  if (deferred.empty())
+    addVarDeclScopesAndTheirAccessors(this);
+  else
+    // Note: the accessors will follow the pattern binding.
+    createSubtreeWithDeferrals<PatternEntryUseScope>(
+        deferred, decl, patternEntryIndex, initializerEnd);
+}
+
+void PatternEntryInitializerScope::expandMe(DeferredNodes &) {
+  DeferredNodes noDeferrals(__FILE__, __LINE__, *this, this);
+  // Create a child for the initializer expression.
+  ScopeCreator().visitExpr(getPatternEntry().getInitAsWritten(), this,
+                           noDeferrals);
+}
+
+void PatternEntryUseScope::expandMe(DeferredNodes &deferred) {
+  // Add accessors for the variables in this pattern.
+  addVarDeclScopesAndTheirAccessors(this);
+  // Create a child for the next pattern binding.
+  createRestOfNestedPatternScopes(decl, deferred, patternEntryIndex + 1);
+}
+
+void ConditionalClauseScope::expandMe(DeferredNodes &deferred) {
+  // If this is a boolean conditional not in a guard continuation, add a
+  // child for the expression.
+  createSubtreeForCondition();
+  // If there's another conditional clause, add it as the child.
+  if (index + 1 < getContainingStatement()->getCond().size())
+    createSubtreeForNextConditionalClause(deferred);
+  else {
```

Comment:
Everywhere else you did if () { } else { }. Can you be consistent with that here.

Reply:
Yes, I see where I used braces but should not have. Thanks. I'm following a guideline from @jordan, which I quite like:
If the statement in the `if` (etc) is visually short (typ. one line) then braces just add clutter.
But if the statement wraps, then braces are better.

Reply:
I'll go through the code and eliminate bogus braces.

Reply:
All I want is for you to be consistent about it.

Reply:
Quite right and fair!